src/neo/Cryptography/MPT/MPTTrie.cs

```diff
@@ -9,22 +10,25 @@ public partial class MPTTrie<TKey, TValue>
         where TValue : class, ISerializable, new()
     {
         private const byte Prefix = 0xf0;
-
+        private bool full;
         private readonly ISnapshot store;
+        private readonly DataCache<UInt256, ByteArrayWrapper> cache;
         private MPTNode root;
 
         public MPTNode Root => root;
 
-        public MPTTrie(ISnapshot store, UInt256 root)
+        public MPTTrie(ISnapshot store, UInt256 root, bool full_state = false)
         {
             this.store = store ?? throw new ArgumentNullException();
-            this.root = root is null ? HashNode.EmptyNode : new HashNode(root);
+            this.cache = new StoreDataCache<UInt256, ByteArrayWrapper>(store, Prefix);
```

Comment:
Is there any limit for this cache?

Reply:
Didn't test sync time yet. I will test in neo3 testnet later when my modification finished.
There is no limit in the cache. It's a `Dictionary`.

Reply:
> There is no limit in the cache.

So it's like holding whole MPT in memory?

Reply:
Not the whole MPT. Only read and write parts in every block.